tensorflow/compiler/xla/service/buffer_assignment.cc

```diff
@@ -1500,20 +1511,25 @@ void BufferAssigner::AssignBuffersFromHeapSimulator(
   }
   VLOG(1) << "Result size from heap simulator: " << result.heap_size;
 
-  BufferAllocation* allocation =
-      assignment->NewEmptyAllocation(result.heap_size, color);
-  for (const auto& buffer_chunk : result.chunk_map) {
-    const HloValue& value = *buffer_chunk.first;
-    const HeapSimulator::Chunk& chunk = buffer_chunk.second;
-    assignment->AddAssignment(allocation, value, chunk.offset, chunk.size);
-  }
-  allocation->peak_buffers_ =
-      ComputePeakMemoryLogicalBuffers(*allocation, result.debug_trace);
+  for (auto& heap_result : result.heap_results) {
+    BufferAllocation* allocation =
+        assignment->NewEmptyAllocation(heap_result.heap_size, color);
+    for (const auto& buffer_chunk : heap_result.chunk_map) {
+      const HloValue& value = *buffer_chunk.first;
+      const HeapSimulator::Chunk& chunk = buffer_chunk.second;
+      assignment->AddAssignment(allocation, value, chunk.offset, chunk.size);
+    }
+    // Compute peak_buffers only when the multiheap mode is off. Simply return
+    // an empty vector in the multiheap mode.
+    if (assignment->multiheap_size_constraint_per_heap() == -1) {
```

Comment:
If possible, I'd really prefer less branching, especially on magic values. Would it be bad to simply compute peak_buffers in both cases?

Reply:
The peak_memory per allocation makes less sense since the peak memory now is across multiple allocations. It is still fine to update the `ComputePeakMemoryLogicalBuffers` to compute the peak memory per allocation though.

I did not revise the `ComputePeakMemoryLogicalBuffers` because it seems only used in unit-tests. Let me check again if I can revise it in a reasonable way.

Reply:
Will revise this `ComputePeakMemoryLogicalBuffers` later. It is not trivial.

Reply:
Done.